Gems/Atom/RPI/Code/Include/Atom/RPI.Edit/Common/JsonUtils.h

```diff
@@ -119,6 +123,62 @@ namespace AZ
                 return false;
             }
 
+            using DeserializedCpuData = AZStd::vector<RHI::CpuProfilingStatisticsSerializer::CpuProfilingStatisticsSerializerEntry>;
+            inline Outcome<DeserializedCpuData, AZStd::string> LoadSavedCpuProfilingStatistics(const AZStd::string& capturePath)
+            {
+                const auto* base = IO::FileIOBase::GetInstance();
+
+                char resolvedPath[IO::MaxPathLength];
+                base->ResolvePath(capturePath.c_str(), resolvedPath, IO::MaxPathLength);
+
+                // NOTE: this uses raw file pointers over the abstractions and utility functions provided by AZ::JsonSerializationUtils because
+                // saved profiling captures can be upwards of 400 MB. This necessitates a buffered approach to avoid allocating huge chunks of memory.
+                FILE* fp = nullptr;
+                azfopen(&fp, resolvedPath, "rb");
+                if (!fp)
+                {
+                    return Failure(AZStd::string::format("Could not fopen file %s, is the path correct?\n", resolvedPath));
+                }
+
+                constexpr AZStd::size_t BufSize = 65536;
```

Comment:
Use AZStd::numeric_limits<size_t>::max() for constants.

Reply:
Is there a way to make this dynamic? I imagine reading a small file will take 65536 bytes whether or not its needed.

Reply:
> Use AZStd::numeric_limits<size_t>::max() for constants.

Could you clarify here? This isn't meant to be the max value of a `size_t`, just a large enough buffer to work well with cache and not malloc too much contiguous memory.

> Is there a way to make this dynamic? I imagine reading a small file will take 65536 bytes whether or not its needed.

An `AZStd::min(fileSize, MaxBufSize)` should work well here - only allocate a buffer equal to the file's size or a predetermined limit. There might be an off-by-one here depending on if rapidjson reads extra bytes for some reason, worst case the first few bytes of the buffer get used twice.

Reply:
Ah yes, since 65536 ==  AZStd::numeric_limits<size_t>::max() i thought it would be good to use that, although this works too just fine.